src/widgets/DatePickerButton.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import React from 'react';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import PropTypes from 'prop-types';
+
+import { CircleButton } from './CircleButton';
+import { CalendarIcon } from './icons';
+
+import { useDatePicker } from '../hooks/useDatePicker';
+
+export const DatePickerButton = ({ initialValue, onDateChanged, maximumDate, minimumDate }) => {
+  const [datePickerIsOpen, openDatePicker, datePickerCallback] = useDatePicker(onDateChanged);
+  return (
+    <>
+      <CircleButton IconComponent={CalendarIcon} onPress={openDatePicker} />
+      {datePickerIsOpen && (
+        <DateTimePicker
+          maximumDate={maximumDate}
+          minimumDate={minimumDate}
+          onChange={datePickerCallback}
+          mode="date"
+          display="spinner"
+          value={initialValue}
+        />
+      )}
+    </>
+  );
+};
```

Comment:
should be doing this more - i.e. when I used the date picker for picking date of births in the `FormControl`/`FormDateInput` component. Try to break components down into the smallest forms and compose them